src/new-components/buttons.tsx

```diff
@@ -60,11 +60,11 @@ LinkButton.defaultProps = defaultButtonProps
 export const PureLinkButton = Button.withComponent('a')
 PureLinkButton.defaultProps = defaultButtonProps
 
-export const TextButton = styled.button`
+export const TextButton = styled.button<{ color?: string }>`
```

Comment:
Should we allow any color here or something like 'just one of the entries in `colorsv2`?

And also can we extract it? 😉

Reply:
I extracted it, but since the types of `colorsV2` are simply for example `violet700: string`, trying to constrain the color type like `typeof colorsV2[keyof typeof colorsV2]` simply becomes `string` so that's no help. 😞

Reply:
Stupid typescript 😞